File: StopWatchAppXamarin/StopWatchApp.Android/Views/LapActivity.cs
using System;
using Android.App;
using StopWatchApp.Core.ViewModels;
using System.Reactive.Disposables;
using Android.OS;
using StopWatchApp.Core.Models;
using Java.Text;
using System.Linq;
using System.Reactive.Linq;
using Locale = Java.Util.Locale;
using Java.Util;
using Reactive.Bindings;
using Android.Widget;
using Reactive.Bindings.Extensions;
using StopWatchApp.Android.Extensions;
using StopWatchApp.Core.Extensions;

namespace StopWatchApp.Android.Views
{
	[Activity (Label = "StopWatchAppXamarin")]
	public class LapActivity : Activity
	{
		private /* readonly */ LapViewModel _viewModel;

		private readonly CompositeDisposable _subscriptionOnCreate = new CompositeDisposable();

		protected override void OnCreate (Bundle savedInstanceState)
		{
			base.OnCreate (savedInstanceState);

			_viewModel = new LapViewModel(this.Application as IModelPool);

			// Set our view from the "main" layout resource
			SetContentView (Resource.Layout.activity_lap);

            // ListView(listLaps, ArrayAdapter) のバインド
            // 番号付きのリストに変換
            var numberedLaps = _viewModel.FormattedLaps.ToNumberedLaps();

			var listLaps = FindViewById<ListView>(Resource.Id.listLaps);
			var listAdapter = new ArrayAdapter(this, global::Android.Resource.Layout.SimpleListItem1);
			listLaps.Adapter = listAdapter;
			listAdapter
				.SetBinding(numberedLaps)
				.AddTo(_subscriptionOnCreate);
		}

		protected override void OnDestroy ()
		{
			// unsubscribe しないと Activity がリークするよ
			_subscriptionOnCreate.Dispose();
			_subscriptionOnCreate.Clear();

			if (_viewModel != null) {
				_viewModel.Dispose ();
			}

			base.OnDestroy ();
		}
	}}

File: StopWatchAppAndroid/app/src/main/java/com/amay077/stopwatchapp/views/activities/LapActivity.java
package com.amay077.stopwatchapp.views.activities;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.amay077.stopwatchapp.App;
import com.amay077.stopwatchapp.R;
import com.amay077.stopwatchapp.databinding.ActivityLapBinding;
import com.amay077.stopwatchapp.viewmodel.LapViewModel;

import javax.inject.Inject;

public class LapActivity extends AppCompatActivity {

    @Inject
    /*private final*/ LapViewModel _viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Inject by Dagger2
        final App app = (App) getApplication();
        app.getApplicationComponent().inject(this);

        ActivityLapBinding binding =  DataBindingUtil.setContentView(this, R.layout.activity_lap);
        binding.setViewModel(_viewModel);
    }

    @Override
    protected void onDestroy() {
        _viewModel.dispose();
        super.onDestroy();
    }
}
